app/assets/javascripts/controllers/middleware_topology/middleware_topology_controller.js

```diff
@@ -1,170 +1,192 @@
-            return self.dblclick(d);
+  };
+
+  $scope.checkboxModel = {
+    value: false
+  };
+  $scope.legendTooltip = 'Click here to show/hide entities of this type';
+
+  $scope.show_hide_names = function() {
+    var vertices = $scope.vs;
+
+    if ($scope.checkboxModel.value) {
+      vertices.selectAll("text.attached-label")
+          .classed("visible", true);
+    } else {
+      vertices.selectAll("text.attached-label")
+          .classed("visible", false);
+    }
+  };
+
+  $scope.refresh();
+  var promise = $interval($scope.refresh, 1000 * 60 * 3);
+
+  $scope.$on('$destroy', function() {
+    $interval.cancel(promise);
+  });
+
+  $scope.$on('render', function(ev, vertices, added) {
+    /*
+     * We are passed two selections of <g> elements:
+     * vertices: All the elements
+     * added: Just the ones that were added
+     */
+    added.attr('class', function(d) {
+      return d.item.kind;
+    });
+    added.append('circle')
+        .attr('r', function(d) {
+          return self.getCircleDimensions(d).r;
+        })
+        .attr('class', function(d) {
+          return topologyService.getItemStatusClass(d);
         });
-        added.append('image')
-             .attr('xlink:href', function (d) {
-                 var iconInfo = self.getIcon(d);
-                 return (iconInfo.type == 'glyph' ? "" : iconInfo.icon);
-             })
-             .attr('y', function (d) {
-                 return self.getCircleDimensions(d).y;
-             })
-             .attr('x', function (d) {
-                 return self.getCircleDimensions(d).x;
-             })
-             .attr('height', function (d) {
-                 return self.getCircleDimensions(d).height;
-             })
-             .attr('width', function(d) {
-                 return self.getCircleDimensions(d).width;
-             });
-
-        // attached labels
-        added.append('text')
-             .attr('x', 26)
-             .attr('y', 24)
-             .text(function(d) {
-                 return d.item.name;
-             })
-             .attr('class', 'attached-label' + ($scope.checkboxModel.value ? ' visible' : ''));
-
-        // possible glyphs
-        added.append('text')
-             .each(function(d) {
-                 var iconInfo = self.getIcon(d);
-                 if (iconInfo.type != 'glyph') {
-                     return;
-                 }
-                 $(this).text(iconInfo.icon)
-                        .attr('class', 'glyph')
-                        .attr('style', 'font-family:' + iconInfo.fontfamily + ';')
-                        .attr('x', 0)
-                        .attr('y', 8);
-             })
-
-
-        added.selectAll('title').text(function(d) {
-            return topologyService.tooltip(d).join('\n');
+    added.append('title');
+    added.on('dblclick', function(d) {
+      return self.dblclick(d);
+    });
+    added.append('image')
+        .attr('xlink:href', function(d) {
```

Comment:
Right, the method call chains are quite tricky
As I understand it correctly on 
https://github.com/airbnb/javascript#whitespace--chains

The indentation depends on what we are calling the methods. If everything is called on the same object (like in a builder pattern) it should be on the same lvl, but it may be also a nested call on a different object... hmm, it's hard to describe, but even harder (or impossible) to check it programmatically.

Hopefully, I've fixed that.